chat-sdk-ui/src/main/java/co/chatsdk/ui/ImageListAdapter.java

```diff
@@ -0,0 +1,107 @@
+package co.chatsdk.ui;
+
+import android.net.Uri;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+
+public class ImageListAdapter extends RecyclerView.Adapter<ImageListAdapter.MyViewHolder>  {
+
+    public ArrayList<Uri> uriArrayList = new ArrayList<>();
+    private WeakReference<SendMultipleImagesAtOnceActivity> theMainActivity;
+
+    public ImageListAdapter (SendMultipleImagesAtOnceActivity activity) {
+        this.theMainActivity = new WeakReference<SendMultipleImagesAtOnceActivity>(activity);
+    }
+
+    public class MyViewHolder extends RecyclerView.ViewHolder {
```

Comment:
Can you move the view holder class to it's own class file. And change it's name to `ImageGalleryViewHolder`

Reply:
I changed it It ImageGalleryViewHolder. However, creating a new class for it creates a multitude of problems. The Programs claims that the ImageGalleryViewHolder class is static, despite the cat that it is not declared as such, and thus it cannot reference the methods in the ImageListAdapter. The ImageListAdpater also states that it must be declared abstract or implement the method onBindViewHolder, in spite of the fact that that method is actually present within that class.